File: src/main/java/eu/mshade/enderframe/world/block/DiggingStatus.java
package eu.mshade.enderframe.world.block;

import java.util.HashMap;
import java.util.Map;

public enum DiggingStatus {

    STARTED(0),
    CANCELLED(1),
    FINISHED(2),
    DROP_ITEM_STACK(3),
    DROP_ITEM(4),
    SHOOT_ARROW_OR_FINISH_EATING(5);

    private int id;
    private final static Map<Integer, DiggingStatus> DIGGING_STATUS_BY_ID = new HashMap<>();

    static {
        for (DiggingStatus diggingStatus : DiggingStatus.values()) {
            DIGGING_STATUS_BY_ID.put(diggingStatus.getId(), diggingStatus);
        }
    }
    DiggingStatus(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public static DiggingStatus fromId(int id){
        return DIGGING_STATUS_BY_ID.get(id);
    }
}

File: src/main/java/eu/mshade/enderframe/world/block/DiggingStatus.kt
package eu.mshade.enderframe.world.block

enum class DiggingStatus(val id: Int) {

    STARTED(0),
    CANCELLED(1),
    FINISHED(2),
    DROP_ITEM_STACK(3),
    DROP_ITEM(4),
    SHOOT_ARROW_OR_FINISH_EATING(5);

    companion object {
        private val DIGGING_STATUS_BY_ID: MutableMap<Int, DiggingStatus> = HashMap()

        init {
            for (diggingStatus in values()) {
                DIGGING_STATUS_BY_ID[diggingStatus.id] = diggingStatus
            }
        }

        fun fromId(id: Int): DiggingStatus? {
            return DIGGING_STATUS_BY_ID[id]
        }
    }
}
